### PR TITLE
Upgraded to latest resource management client and batch management lib

### PR DESCRIPTION
**Pull request by brnleehng**

### CODE REVIEW

File: CSharp/AccountManagement/AccountManagement.csproj [hunk 1]
```diff
@@ -61,6 +61,14 @@
...
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>packages\Microsoft.Rest.ClientRuntime.2.2.0\lib\net45\Microsoft.Rest.ClientRuntime.dll</HintPath>
+      <SpecificVersion>False</SpecificVersion>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime.Azure, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>packages\Microsoft.Rest.ClientRuntime.Azure.3.2.0\lib\net45\Microsoft.Rest.ClientRuntime.Azure.dll</HintPath>
+      <SpecificVersion>False</SpecificVersion>
+    </Reference>
     <Reference Include="Microsoft.Threading.Tasks, Version=1.0.12.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
```
Comment (jasper-schneider) [inline CSharp/AccountManagement/AccountManagement.csproj]: With the move to the AutoRest proxies for both Batch and Resources, is this package still needed?

File: CSharp/AccountManagement/Program.cs [hunk 2]
```diff
@@ -207,21 +221,21 @@ private static string PromptUserForAccountName()
...
+        /// <param name="resourceManagementClient">The <see cref="Microsoft.Azure.Management.ResourceManager.IResouceManagementClient"/>
         /// to use when creating the resource group.</param>
         /// <param name="location">The location where the resource group will be created.</param>
         /// <returns>A <see cref="System.Threading.Tasks.Task"/> object that represents the asynchronous operation.</returns>
         private static async Task CreateResourceGroupAsync(IResourceManagementClient resourceManagementClient, string location)
         {
-            ResourceGroupExistsResult existsResult = await resourceManagementClient.ResourceGroups.CheckExistenceAsync(ResourceGroupName);
-            if (!existsResult.Exists)
+            bool? existsResult = await resourceManagementClient.ResourceGroups.CheckExistenceAsync(ResourceGroupName);
+            if (!existsResult.Value)
```
Comment (jasper-schneider) [inline CSharp/AccountManagement/Program.cs]: Should probably check for null before accessing the .Value property.  It's a little odd to me that a 3rd state exists at all - either the group exists or it doesn't, so what does the third state even mean?  But yeah, the check should probably be "if (existsResult == null || !existsResult.Value)"

File: CSharp/AccountManagement/Program.cs [hunk 3]
```diff
@@ -249,30 +263,31 @@ private static string PromptUserForAccountName()
         /// <summary>
         /// Performs various Batch account operations using the Batch Management library.
         /// </summary>
-        /// <param name="creds">The <see cref="Microsoft.Azure.TokenCloudCredentials"/> containing information about the user's
-        /// Azure account and subscription.</param>
+        /// <param name="accessToken">The access token to use for authentication.</param>
+        /// <param name="subscriptionId">The subscription id to use for creating Batch management client</param>
```
Comment (jasper-schneider) [inline CSharp/AccountManagement/Program.cs]: creating _**the**_ Batch management client

File: CSharp/AccountManagement/Program.cs [hunk 4]
```diff
@@ -249,30 +263,31 @@ private static string PromptUserForAccountName()
...
         /// Performs various Batch account operations using the Batch Management library.
         /// </summary>
-        /// <param name="creds">The <see cref="Microsoft.Azure.TokenCloudCredentials"/> containing information about the user's
-        /// Azure account and subscription.</param>
+        /// <param name="accessToken">The access token to use for authentication.</param>
+        /// <param name="subscriptionId">The subscription id to use for creating Batch management client</param>
         /// <param name="location">The location where the Batch account will be created.</param>
         /// <returns>A <see cref="System.Threading.Tasks.Task"/> object that represents the asynchronous operation.</returns>
-        private static async Task PerformBatchAccountOperationsAsync(TokenCloudCredentials creds, string location)
+        private static async Task PerformBatchAccountOperationsAsync(string accessToken, string subscriptionId, string location)
```
Comment (jasper-schneider) [inline CSharp/AccountManagement/Program.cs]: Is the subscriptionId parameter actually used?  Line 272 only uses the accessToken.

File: CSharp/AccountManagement/packages.config [hunk 5]
```diff
@@ -3,12 +3,14 @@
   <package id="Hyak.Common" version="1.0.3" targetFramework="net45" />
   <package id="Microsoft.Azure.Common" version="2.1.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Common.Dependencies" version="1.0.0" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.Batch" version="1.5.0" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.Resources" version="2.18.1-preview" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.Batch" version="2.0.0" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.ResourceManager" version="1.1.1-preview" targetFramework="net45" />
   <package id="Microsoft.Bcl" version="1.1.9" targetFramework="net45" />
```
Comment (jasper-schneider) [inline CSharp/AccountManagement/packages.config]: Same, do we still need the Bcl packages and the Hyak.Common package now that we've moved to the AutoRest proxies?

File: CSharp/AccountManagement/Program.cs (hunk 3 above)
Comment (brnleehng) [inline CSharp/AccountManagement/Program.cs]: Fixed

File: CSharp/AccountManagement/Program.cs (hunk 4 above)
Comment (brnleehng) [inline CSharp/AccountManagement/Program.cs]: On Line 274, the subscription id is set for the batch management client. Subscription id is a required parameter in the swagger spec that needs to be set first before calling any batch account oeprations

File: CSharp/AccountManagement/packages.config (hunk 5 above)
Comment (brnleehng) [inline CSharp/AccountManagement/packages.config]: Yes, there are test dependencies in the sdk-for-net repo that still have the Hyak.Common package and Bcl package which also makes our package dependent on these packages

File: CSharp/AccountManagement/Program.cs (hunk 4 above)
Comment (jasper-schneider) [inline CSharp/AccountManagement/Program.cs]: Got it, I missed that :)

File: CSharp/AccountManagement/packages.config (hunk 5 above)
Comment (jasper-schneider) [inline CSharp/AccountManagement/packages.config]: All right.  That's unfortunate, but we have to live with it for a while longer.

File: CSharp/AccountManagement/Program.cs (hunk 2 above)
Comment (brnleehng) [inline CSharp/AccountManagement/Program.cs]: I'll add this check. This is the Resource Management Client, so I'll take a look into why they have a third state.

File: CSharp/AccountManagement/AccountManagement.csproj (hunk 1 above)
Comment (brnleehng) [inline CSharp/AccountManagement/AccountManagement.csproj]: Yes, test dependencies in the sdk-for-net still requires this.

**Pull request merged**